client/app/components/pages/workspace/annotation.component.ts

```diff
@@ -33,21 +35,45 @@ export class AnnotationComponent implements OnInit {
 	isPainting:boolean;
 	ctx: CanvasRenderingContext2D;
 	currentPointInDraw: any;
+	currentFreeDrawLineText :String;
+	currentFreeDrawLineNum: number;
+	allFreeDrawLines : [FreeDraw]
+	currentFreeDrawLine : FreeDraw
 	
-
 	constructor(private windowService: WindowService, private manuscriptsService: ManuscriptsService){
 		this._window = windowService.nativeWindow;
 	}
 
+	saveFreeDrawLine(){
```

Comment:
I think binding the this.currentFreeDrawLine.text to the textbox will be more comfortable UXly.